harness/determined/util.py

```diff
@@ -210,3 +211,22 @@ def filter_duplicates(
             duplicates.add(item)
         last_item = item
     return duplicates
+
+
+T = TypeVar("T", bound=Callable[..., Any])
+
+
+def deprecated(msg: str) -> Callable[[T], T]:
+    def make_wrapper(fn: T) -> T:
+        warned = False
+
+        def wrapper(*arg: List, **kwarg: Dict) -> Any:
+            nonlocal warned
+            if not warned:
```

Comment:
`warnings.warn(msg, FutureWarning)` is probably more pythonic way to report deprecations, and it automatically takes care of logging the message only once.

Reply:
Great!  I didn't know about that.